### PR TITLE
Store Emails

### PR DESCRIPTION
**Pull request by Kukks**
This PR starts adding email sending capabilities based on invoice events.

### CODE REVIEW

**Comment by dstrukt** [comment]
Love that this is being worked on, hyped to see it in action!

**Comment by dennisreimann** [comment]
I've updated the email rules layout, so that @dstrukt can do further modification mockups based on this version.

Please note that the "Customer email" checkbox works in addition to the "Recipients" input. The recipients are optional and if the checkbox is active, the customer will be added to the recipients (or receive the email as only one, if there are no recipients).

grafik

**Comment by dstrukt** [comment]
Oooo that's already looking quite nice, great work, and thank you!

**Comment by dennisreimann** [comment]
@dstrukt I think we should get this reviewed and merged, then you can polish it in a separate PR.

**Review by dennisreimann — APPROVED** [review]
tACK. Looks good to me.

**Comment by dstrukt** [comment]
tACK, finally looked at the cleaned up UI, much nicer :) 

And agree @dennisreimann, send it!

**Review by dstrukt — APPROVED** [review]

File: BTCPayServer/Controllers/UIStoresController.Email.cs [hunk 1]
```diff
@@ -47,42 +130,36 @@ public async Task<IActionResult> Emails(string storeId, EmailsViewModel model, s
...
                 store.SetStoreBlob(storeBlob);
                 await _Repo.UpdateStore(store);
                 TempData[WellKnownTempData.SuccessMessage] = "Email server password reset";
-                return RedirectToAction(nameof(Emails), new
-                {
-                    storeId
-                });
+                return RedirectToAction(nameof(StoreEmailSettings), new { storeId });
             }
-            else // if(command == "Save")
```
Comment (NicolasDorier) [inline BTCPayServer/Controllers/UIStoresController.Email.cs:70]: why you remove those comments?

File: BTCPayServer/Controllers/UIServerController.cs [hunk 2]
```diff
@@ -1038,33 +1037,32 @@ public async Task<IActionResult> Emails(EmailsViewModel model, string command)
...
+            
+            if (command == "ResetPassword")
             {
                 var settings = await _SettingsRepository.GetSettingAsync<EmailSettings>() ?? new EmailSettings();
                 settings.Password = null;
                 await _SettingsRepository.UpdateSetting(model.Settings);
                 TempData[WellKnownTempData.SuccessMessage] = "Email server password reset";
                 return RedirectToAction(nameof(Emails));
             }
-            else // if(command == "Save")
```
Comment (NicolasDorier) [inline BTCPayServer/Controllers/UIServerController.cs:1057]: why you remove those comments?

**Comment by NicolasDorier** [comment]
This PR doesn't work. The "Email rules" in the store's email redirect to 404.
Add a small test in selenium.

**Comment by NicolasDorier** [comment]
imho, `Email Rules` should show up in the store's settings even if the settings aren't complete.

**Comment by dennisreimann** [comment]
@NicolasDorier Updated. 

In case the settings arn't complete, we'll display a warning on the rules configuration page.

grafik

**Comment by NicolasDorier** [comment]
My last commit `[68a1d4b](https://github.com/btcpayserver/btcpayserver/pull/3611/commits/68a1d4b0d9a106d0671cf63d0cb214d9f9d7fdcd) made it so all the time we required an email, we now accept `Nicolas Dorier <blah@example.com>` as valid input. This will make emails nicer.

**Pull request merged**